iroha/src/sumeragi.rs

```diff
@@ -68,45 +83,74 @@ impl Sumeragi {
         }
     }
 
-    pub fn peer_id(&self) -> &PeerId {
-        &self.peer_id
-    }
-
-    pub fn has_pending_block(&self) -> bool {
-        self.pending_block.is_some()
+    /// the leader of each round just uses the transactions they have at hand to create a block
+    pub async fn round(&mut self, transactions: Vec<Transaction>) -> Result<Option<Block>, String> {
+        if let Role::Leader = self.role() {
+            let block = Block::builder(
+                transactions
+                    .into_iter()
+                    .map(|tx| tx.sign(Vec::new()))
+                    .filter_map(Result::ok)
+                    .collect(),
+            )
+            .build()
+            .sign(&self.public_key, &self.private_key)?;
+            let minimum_quorum_of_peers = 2;
+            if self.sorted_peers.len() < minimum_quorum_of_peers {
+                Ok(Some(block))
```

Comment:
Maybe it should also `kura.store` the block inside of the sumeragi. This behavior is a bit misleading: if there is only one peer it returns the block and stores it in `lib.rs`, if more peers it stores peer in `sumeragi.rs` .

Reply:
https://soramitsu.atlassian.net/browse/RH2-87